pkg/apis/calico/validation/networkconfig.go

```diff
@@ -101,28 +104,36 @@ func ValidateNetworkConfigIPV4(ipv4 *apiscalico.IPv4, fldPath *field.Path) field
 // ValidateNetworkConfigIPV6 validates the IPv6 configuration in the network config.
 func ValidateNetworkConfigIPV6(ipv6 *apiscalico.IPv6, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+
 	if ipv6 == nil {
 		return allErrs
 	}
 
-	if ipv6.Pool != nil && *ipv6.Pool != "" && !sets.New(apiscalico.PoolIPIP, apiscalico.PoolVXLan).Has(*ipv6.Pool) {
+	if ipv6.Pool != nil && !sets.New(apiscalico.PoolIPIP, apiscalico.PoolVXLan).Has(*ipv6.Pool) {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("pool"), *ipv6.Pool, fmt.Sprintf("unsupported value %q for pool, supported values are [%q, %q]", *ipv6.Pool, apiscalico.PoolIPIP, apiscalico.PoolVXLan)))
 	}
 
 	if ipv6.Mode != nil && !sets.New(apiscalico.Always, apiscalico.Never, apiscalico.CrossSubnet, apiscalico.Off).Has(*ipv6.Mode) {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("mode"), *ipv6.Mode, fmt.Sprintf("unsupported value %q for mode, supported values are [%q, %q, %q, %q]", *ipv6.Mode, apiscalico.Always, apiscalico.Never, apiscalico.CrossSubnet, apiscalico.Off)))
 	}
 
+	// Check for unsupported mode with VXLan pool
+	// VXLan pool only supports Always and Never modes, so if the mode is set
+	// to CrossSubnet or Off, it is invalid.
+	if ipv6.Pool != nil && *ipv6.Pool == apiscalico.PoolVXLan && !sets.New(apiscalico.Always, apiscalico.Never).Has(*ipv6.Mode) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("mode"), *ipv6.Mode, fmt.Sprintf("unsupported value %q for mode with pool %q, supported values are [%q, %q]", *ipv6.Mode, apiscalico.PoolVXLan, apiscalico.Always, apiscalico.Never)))
+	}
+
 	if ipv6.AutoDetectionMethod != nil && *ipv6.AutoDetectionMethod != "" {
-		allErrs = append(allErrs, ValidateIPAutodetectionMethod(*ipv6.AutoDetectionMethod, fldPath.Child("autoDetectionMethod"))...)
+		allErrs = append(allErrs, ValidateIPAutoDetectionMethod(*ipv6.AutoDetectionMethod, fldPath.Child("autoDetectionMethod"))...)
 	}
 
 	return allErrs
 }
 
-func ValidateIPAutodetectionMethod(method string, fldPath *field.Path) field.ErrorList {
+func ValidateIPAutoDetectionMethod(method string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	validOptions := sets.New("first-found", "can-reach", "interface", "skip-interface", "cidr")
+	validOptions := sets.New("first-found", "can-reach", "interface", "skip-interface", "cidr", "kubernetes-internal-ip")
```

Comment:
Let's add a validation that `len(parts) == 1` for `first-found` and `kubernetes-internal-ip`.